parsons/databases/mysql/create_table.py

```diff
@@ -0,0 +1,159 @@
+import ast
+import petl
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MySQLCreateTable():
+
+    def __init__(self):
+
+        pass
+
+    def data_type(self, val, current_type):
+        # Evaluate the MySQL data type of a given value. Then return
+        # the data type of the value, while also weighing what the
+        # current type is.
+
+        # Stop if the current type is already a varchar
+        if current_type == 'varchar':
+            return current_type
+
+        # Convert to string to reevaluate data type
+        try:
+            t = ast.literal_eval(str(val))
+        except ValueError:
+            return 'varchar'
+        except SyntaxError:
+            return 'varchar'
+
+        # If a float, then end
+        if type(t) == float:
+            return 'float'
+
+        # If an int and current type is not a float, then determine type of int.
+        if type(t) == int and current_type != 'float':
+            # Make sure that it is a valid integer
+            if not self.is_valid_integer(val):
+                return 'varchar'
+            # Use smallest possible int type above TINYINT
+            if (-32768 < t < 32767) and current_type not in ['int', 'bigint', 'mediumint']:
+                return 'smallint'
+            elif (-8388608 < t < 8388607) and current_type not in ['int', 'bigint']:
+                return 'mediumint'
+            elif (-2147483648 < t < 2147483647) and current_type not in ['bigint']:
+                return 'int'
+            else:
+                return 'bigint'
+
+        # Finally return the current type
+        else:
+            return current_type
+
+    def is_valid_integer(self, val):
+
+        # Valid ints in python can contain an underscore, but Redshift can't. This
+        # checks to see if there is an underscore in the value and turns it into
+        # a varchar if so.
+        try:
+            if '_' in val:
+                return False
+
+        except TypeError:
+            return True
+
+        # If it has a leading zero, we should treat it as a varchar, since it is
+        # probably there for a good reason (e.g. zipcode)
+        if val.isdigit():
+            if val[0] == '0':
+                return False
+
+        return True
+
+    def evaluate_column(self, column_rows):
+        # Generate MySQL data types and widths for a column.
+
+        col_width = 0
+        col_type = None
+
+        # Iterate through each row in the column
+        for row in column_rows:
+
+            # Get the MySQL data type
+            col_type = self.data_type(row, col_type)
+
+            # Calculate width if a varchar
+            if col_type == 'varchar':
+                row_width = len(str(row.encode('utf-8')))
+
+                # Evaluate width vs. current max width
+                if row_width > col_width:
+                    col_width = row_width
+
+        return col_type, col_width
```

Comment:
Maybe we should default to making the column width wider than the existing maximum for rows already in place?